edgelet/edgelet-kube/src/runtime.rs

```diff
@@ -181,19 +194,53 @@ impl KubeModuleRuntime<HttpClient<HttpsConnector<HttpConnector>, Body>> {
     }
 }
 
-impl<S> ModuleRegistry for KubeModuleRuntime<S>
+impl KubeModuleRuntime<ValueToken, HttpClient<HttpsConnector<HttpConnector>, Body>> {
```

Comment:
Curious: why is this factoring of this code better? i.e., have a `with_client` and a `new`? Is it for making it easier to test?

Reply:
I need `with_client` method to substitute `reqwest` with any service for example for tests. Because KubeModuleRuntime is not used anywhere so far, I can easily change `with_client` -> `new(client, ...)` and delete old `new()` implementation

Reply:
Got it. Yeah, that's what I thought.